### ISSUE TITLE
up-sampled imbalanced samples

### ISSUE THREAD

**Issue by FSantosCodes** [imbs-hl/ranger#412]
Dear ranger community,
I experienced a low performance (>15 min.) when I apply ranger for a classification problem using caret::upSample to balance my dataset (which has ~400 observations but after upsampling it end with ~1200 observations). I identified that this happens when I upsample classes that have 1 or 2 observations for one class but in others more than 20 observations. As I cannot control these dataset conditions (I am running ranger several times), it took me time to found a minimum number of observations (>= 5) for ranger work fine. Therefore, I think could be nice to have a warning or note in the manual indicating a minimum number of observations that ranger requires in a class to solve a classification problem. Furthermore, I am wondering how to solve the same problem (i.e. unbalanced samples) but for regression analysis, since performance is low, do you have any idea?

Thanks and best regards.

Fabián.

**Comment by mnwright** [comment]
Could you give a reproducible example? Generally, it should run fine on unbalanced samples. You might also use one of the ranger-internal weighting schemes instead of caret to speed things up.

**Comment by FSantosCodes** [comment]
Thanks and sorry for the late reply. I was searching for this dataset as is difficult to track these errors as I am running ranger in parallel. However, after some tests I found that the problem was my PC due a RAM issue or something related to its resources. Therefore, ignore or erase this issue. I added the dataset with these problems and a short script of what I did in any case.
The only thing that I am wondering, why ranger-classification trains fine only if I upsample my dataset? and, what can I do to improve ranger-regression if my dataset is unbalanced?

Best regards,

Fabián.

#Dataset at: https://raw.githubusercontent.com/FSantosCodes/test/celldata.rds
#and run the next lines:
cell.data <- readRDS("celldata.rds")
table(cell.data$Class) #pretty unbalanced
cell.data <- upSample(cell.data[,2:ncol(cell.data)],cell.data[,1],yname=names(cell.data)[1])
table(cell.data$Class) #now balanced
cell.weights <- cell.data$dist #i apply case weights
cell.data <- cell.data[,names(cell.data)!="dist"] #then I remove this weights from databaseto predict
#run ranger
cell.rf <- ranger(formula=Class~.,
                  data=cell.data,
                  replace=F,
                  min.node.size=1,
                  case.weights=cell.weights,
                  scale.permutation.importance=T,
                  importance="permutation")

**Comment by mnwright** [comment]
Still, there shouldn't be a problem with unbalanced data, it might actually be faster because the data could be separated in a few splits. If anyone finds a reproducible example for this problem, please share.
